### DISCUSSION TITLE
Highlight the valid egg moves in the relearn drop down

### DISCUSSION THREAD

**Discussion by ghost** [kwsch/PKHeX#3806] [Ideas]

**Comment by kwsch** [comment]
Relearn moves are very specific to the encounter object it is matched to, and it can be one of many different lists.
You can click on the group box to have it suggest a possible list of relearn moves, as described in the program's shortcut list.

**Reply by kwsch** [reply]
There are many legal states, because you can share egg moves after breeding in future games, or you don't have any moves that are needed in the relearn list.
The moves you selected don't have to be in the relearn list. And, if the program has no issue with your current relearn moves, then you don't /need/ to change them.
Sure it could show every possible relearn move, but some combinations are not possible, or are dependent on the encounter type. It's not really worth it when you can just auto suggest to fill it in; these moves are not visible outside of the move relearner.

**Reply by kwsch** [reply]
It's inadvisable to hide extra (legal) moves in the relearn moves, as they aren't visible unless you're at the move relearner or peeking at the data with a save editor. It will always be faster to just edit the save file / inject new data into the game if you're wanting to change the moveset.

Again, showing a "legal list" is subjective and difficult. Certain moves cannot exist at arbitrary slots, due to how egg move inheritance works. For example, your Ralts may only have Growl at the first move slot since it is a level up move, and any inherited moves must be in the slots after. If you don't pick Growl as the first relearn move, then all other move slots must be non-zero moves as Growl must be pushed out of the list in order to disappear. Starting from a blank slate will have no relearn moves legal, because you must have a correctly ordered set of relearn moves to be legal. If we were to instead just show a list of possible relearn moves, some combinations would not be possible, and the order would still have to be correct based on inheritance. If we just showed egg moves, that would be misleading as other moves can be present, and if egg moves are present, then all 4 moves must be filled in (to remove base levelup moves). Some Pokémon don't have egg moves (like Magikarp), but need to have levelup moves present.

Also, there are encounters like DexNav that get a single egg move, so you can only pick one Egg Move (no other source) for the first slot and all other slots must be zero. Other encounters like Gen7 static gift Eevee are indistinguishable from bred Eevee, except they have no relearn moves. What should be shown in that case?

As you can see, it is a very sticky thing to provide "legality" for, which is why it's heavily suggested to just let the program suggest legal values.

**Reply by Lusamine** [reply]
My impression of this whole discussion is that you feel it's too inconvenient to look things up to generate legal Pokémon, but my opinion is that it's expected of you. The program already holds your hand in an incredible number of ways and you want it to be completely autonomous -- but the truth is, many things people want to edit do require the user to look it up outside.

If it hasn't been made clear to you now through this exchange, it is actually inconvenient to add this elegantly because of the many edge cases and legality checks already present for egg moves. It isn't as simple as you think to just highlight what you want. I would actually object to something that crowds the UI and already has a very viable alternative, which is to look it up.

The program is open source, which means anyone can edit it. If someone agrees with you that it's extremely inconvenient to look it up, and they don't think it's inconvenient to implement the entire lookup in the program, and they can do it elegantly to cover edge case handling (such as a Pokemon that has made it through several games with different lists including ones that let you pass moves without it being in relearns), they can submit a pull request to be reviewed. Likewise, if you think it's simple enough to add, nothing really stops you from forking the repository and adding a limited feature for yourself.
